src/lib/core/NodeId.h

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace chip {
+
+// TODO: Consider making this a class and the various utility methods static
+// methods.
+using NodeId = uint64_t;
+
+constexpr NodeId kUndefinedNodeId = 0ULL;
+constexpr NodeId kAnyNodeId       = 0xFFFF'FFFF'FFFF'FFFFULL;
+
+constexpr bool IsOperationalNodeId(NodeId aNodeId)
+{
+    return (aNodeId != kUndefinedNodeId) && (aNodeId <= 0xFFFF'FFEF'FFFF'FFFFUL);
```

Comment:
Was it intended that this instead be the following?

```suggestion
    return (aNodeId != kUndefinedNodeId) && (aNodeId < kAnyNodeId);
```

If not, can we have a constexpr for `0xFFFF'FFEF'FFFF'FFFFUL`?

Reply:
No, this was definitely intended to be what it is.  I'll add a constexpr for that.